### DISCUSSION TITLE
Is Mui5 with Emotion possible with Server Components?

### DISCUSSION THREAD

**Discussion by chrisb2244** [vercel/next.js#41834] [Help]
With NextJS 12, Mui5 suggests a custom `_document.tsx` and `_app.tsx` as described here: https://github.com/mui/material-ui/tree/master/examples/nextjs-with-typescript

NextJS 13's migration guide instructs us to basically copy `_document` and `_app` into the root layout, but indicates that "If you are using any React Context providers, they will need to be moved to a [Client Component](https://beta.nextjs.org/docs/rendering/server-and-client-components)".

Given that seemingly any use of Mui requires the context provided by Emotion (with the default Emotion styling - there are other options), is it possible to rearrange the structure of my Components or pages in any way to allow use with Server Components? Or does this combination essentially force everything to include `'use client';` or similar?

Is there any hope that this might change (if I haven't just completely misunderstood)? Or should I be looking at the alternative styling options described here: [Mui Interoperability](https://mui.com/material-ui/guides/interoperability/) and here: [Mui styled-engine](https://mui.com/material-ui/guides/styled-engine/) (incidentally, comes with a big warning about SSR) for migration pathways?

**Comment by MasterProvider** [comment]
I have the same problem, any Idea about supporting MUI in NEXTJS 13 I tried to move _document.tsx and _app.tsx  to my layout.tsx but isn't work at all.

**Comment by chrisb2244** [comment]
There's an example here [emotion-js/emotion issue 2928 (Plans to support Next.js 13 - /app directory)](https://github.com/emotion-js/emotion/issues/2928#issuecomment-1293012737) that reportedly provides a working render, although reading the comments it's not clear that it provides the streaming behaviour. 

With the issue with streaming as described in the comment [here](https://github.com/emotion-js/emotion/issues/2928#issuecomment-1294087939), if I've understood correctly it shows CSS that is streamed with components after initial rendering doesn't get applied.
